File: src/main/java/com/langleon/dsobuildsim/wisdomskilltree/dto/definition/WisdomGroupDefinitionDTO.java
package com.langleon.dsobuildsim.wisdomskilltree.dto.definition;

import com.langleon.dsobuildsim.wisdomskilltree.wisdomgroup.WisdomGroupType;
import com.langleon.dsobuildsim.wisdomskilltree.wisdomskill.WisdomSkillType;

import java.util.EnumMap;

public record WisdomGroupDefinitionDTO(
        WisdomGroupType type,
        String name,
        int maxLevel,
        EnumMap<WisdomSkillType, WisdomSkillDefinitionDTO> wisdomSkills
        ) {
}

File: frontend/src/app/models/gamedataDTOs/WisdomGroupDefinitionDTO.ts
import { WisdomGroupType } from "../../enums/WisdomGroupType";
import { WisdomSkillType } from "../../enums/WisdomSkillType";
import { WisdomSkillDefinitionDTO } from "./WisdomSkillDefinitionDTO";

export interface WisdomGroupDefinitionDTO {
    type: WisdomGroupType;
    name: string;
    maxLevel: number;
    wisdomSkills: Record<WisdomSkillType, WisdomSkillDefinitionDTO>;
}